python-mini-projects/projects/Duplicate files remover/duplicatefileremover.py

```diff
@@ -0,0 +1,40 @@
+import hashlib
+import os
+
+# Returns the hash string of the given file name
+
+
+def hashFile(filename):
+    # For large files, if we read it all together it can lead to memory overflow, So we take a blocksize to read at a time
+    BLOCKSIZE = 65536
+    hasher = hashlib.md5()
+    with open(filename, 'rb') as file:
+        # Reads the particular blocksize from file
+        buf = file.read(BLOCKSIZE)
+        while(len(buf) > 0):
+            hasher.update(buf)
+            buf = file.read(BLOCKSIZE)
+    return hasher.hexdigest()
+
+
+if __name__ == "__main__":
+    # Dictionary to store the hash and filename
+    hashMap = {}
+
+    # List to store deleted files
+    deletedFiles = []
+    filelist = [f for f in os.listdir() if os.path.isfile(f)]
+    for f in filelist:
+        key = hashFile(f)
+        # If key already exists, it deletes the file
+        if key in hashMap.keys():
+            deletedFiles.append(f)
+            os.remove(f)
+        else:
+            hashMap[key] = f
+    if len(deletedFiles) == 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)